### PR TITLE
Add Class#subclasses

### PR DESCRIPTION
**Pull request by casperisfine**
Ref: https://bugs.ruby-lang.org/issues/18273

Returns an array containing the receiver's direct subclasses without
singleton classes.

Implements [Feature #18273]

### CODE REVIEW

File: test/ruby/test_class.rb [hunk 1]
```diff
@@ -755,4 +755,26 @@ def test_descendants
     assert_include(object_descendants, sc)
     assert_include(object_descendants, ssc)
   end
+
+  def test_subclasses
+    c = Class.new
+    sc = Class.new(c)
+    ssc = Class.new(sc)
+    p [c, sc, ssc]
```
Comment (eregon) [inline test/ruby/test_class.rb:763]: Debug print left here

Comment (casperisfine) [reply test/ruby/test_class.rb]: Oops. Fixed.

File: include/ruby/internal/intern/class.h [hunk 2]
```diff
@@ -187,6 +187,19 @@ VALUE rb_mod_ancestors(VALUE mod);
...
 VALUE rb_class_descendants(VALUE klass);
 
+/**
+ * Queries the class's direct descendants. This  routine gathers classes that are
+ * direct subclasses of the given class,
+ * returning an array of classes that have the given class as a superclass.
+ * The returned array does not include singleton classes.
+ *
+ * @param[in]  klass A class.
+ * @return     An array of classes where `klass` is the `supeclass`.
```
Comment (paarthmadan) [inline include/ruby/internal/intern/class.h:197]:
```suggestion
 * @return     An array of classes where `klass` is the `superclass`.
```

File: include/ruby/internal/intern/class.h [hunk 3]
```diff
@@ -187,6 +187,19 @@ VALUE rb_mod_ancestors(VALUE mod);
  */
 VALUE rb_class_descendants(VALUE klass);
 
+/**
+ * Queries the class's direct descendants. This  routine gathers classes that are
```
Comment (paarthmadan) [inline include/ruby/internal/intern/class.h:191]:
```suggestion
 * Queries the class's direct descendants. This routine gathers classes that are
```

**Review by paarthmadan — COMMENTED** [review]
Sorry for the drive-by review, stumbled upon this code and noticed some small typos.

**Comment by casperisfine** [comment]
Thanks @paarthmadan !

**Comment by casperisfine** [comment]
The CI failure is from the debug gem: https://github.com/ruby/debug/pull/398

**Comment by byroot** [comment]
Other than the `debug` gem test suite, it seems good to go.

**Review by jeremyevans — APPROVED** [review]
Looks good to me.

**Pull request merged**